src/System.IO/tests/BinaryWriter/BinaryWriterTests.cs

```diff
@@ -0,0 +1,391 @@
+
+            dw2.Dispose();
+            mstr.Dispose();
+        }
+        [TestMethod]
+        public static void BinaryWriter_SeekTests_Negative()
+        {
+            int[] iArrInvalidValues = new Int32[] { -1, -2, -100, -1000, -10000, -100000, -1000000, -10000000, -100000000, -1000000000, Int32.MinValue, Int16.MinValue };
+            Int64 lReturn = 0;
+            
+            // [] ArgumentOutOfRangeException if offset is negative
+            MemoryStream mstr = new MemoryStream();
+            BinaryWriter dw2 = new BinaryWriter(mstr);
+            dw2.Write("Hello, this is my string".ToCharArray());
+            for (int iLoop = 0; iLoop < iArrInvalidValues.Length; iLoop++)
+            {
+                Assert.Throws<IOException>(() => { lReturn = dw2.Seek(iArrInvalidValues[iLoop], SeekOrigin.Begin); }, "Error_17dy7! Expected exception not thrown");
+                Assert.IsTrue(lReturn == 0, "Error_2t8gy! Incorrect return value, lReturn==" + lReturn);
+            }
+            dw2.Dispose();
+            mstr.Dispose();
+
+            // [] ArgumentException for invalid seekOrigin
+            mstr = new MemoryStream();
+            dw2 = new BinaryWriter(mstr);
+            dw2.Write("012345789".ToCharArray());
+            Assert.Throws<ArgumentException>(() => { lReturn = dw2.Seek(3, ~SeekOrigin.Begin); }, "Error_109ux! Expected exception not thrown");
+
+            dw2.Dispose();
+            mstr.Dispose();
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_BaseStreamTests()
+        {
+            // [] Get the base stream for MemoryStream
+            MemoryStream ms2 = new MemoryStream();
+            BinaryWriter sr2 = new BinaryWriter(ms2);
+            if (sr2.BaseStream != ms2)
+                Assert.Fail("Error_g57hb! Incorrect basestream");
+
+            sr2.Dispose();
+            ms2.Dispose();
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_FlushTests()
+        {
+            // [] Check that flush updates the underlying stream
+            MemoryStream memstr2 = new MemoryStream();
+            BinaryWriter bw2 = new BinaryWriter(memstr2);
+
+            bw2.Write("HelloWorld");
+            //TODO:: Ckeck with dev why it's 11 bytes.
+            Assert.AreEqual(11, memstr2.Length, "Error_984y7! Incorrect stream length==" + memstr2.Length);
+            bw2.Flush();
+            Assert.AreEqual(11, memstr2.Length, "Error_8589v! Incorrect stream length after flush==" + memstr2.Length);
+
+            // [] Flushing closed writer should not throw an exception.
+            memstr2 = new MemoryStream();
+            bw2 = new BinaryWriter(memstr2);
+            bw2.Dispose();
+            try
+            {
+                bw2.Flush();
+            }
+            catch (Exception exc)
+            {
+                Assert.Fail("Error_399c;! Unexpected exception exception thrown, exc==" + exc.ToString());
```

Comment:
Rather than catching an asserting, for cases like this we usually just let the exception propagate out of the test.